tests/file_system_cache_test.py

```diff
@@ -122,5 +123,19 @@ def test_file_bigger_than_cache(self):
       cache.put("foo", b"bar")
       self.assertEqual(cache.get("foo"), None)
 
+  def test_threads(self):
+    with tempfile.TemporaryDirectory() as tmpdir:
+      cache = FileSystemCache(tmpdir)
+      for i in range(100):
+          threading.Thread(target=self.call_multiple_puts_and_gets(cache))
+          self.assertEqual(cache.get("foo"), b"bar")
+
+  def call_multiple_puts_and_gets(self, cache):
```

Comment:
Style nit: since this is a one-off function needed for the threads in the above test, you can actually put this function definition inside the `test_threads` function. I put it as the first thing in `test_threads`, like before creating the tmpdir.